cpp/src/arrow/vendored/datetime/tz.h

```diff
@@ -294,6 +294,40 @@ struct zoned_traits
 {
 };
 
+class OffsetZone {
```

Comment:
As a policy, we shouldn't add our own code inside vendored libraries. Just put it somewhere else in the source tree (for example in a `arrow/util/date_internal.h` where the vendored code would be included)

Reply:
Will do.

Reply:
Moved. The code in the file is adapted from [here](https://howardhinnant.github.io/date/tz.html#Examples), but I've added our license header. Is that ok?